### DISCUSSION TITLE
Custom Input: Best way to pass register and error

### DISCUSSION THREAD

**Discussion by ftzi** [react-hook-form/react-hook-form#1864] [Q&A] [answered]
Hey there! Very nice lib.

I have a custom input component which also receives the error message. 

Here is how it is the calling tag currently:

`<Input name="name" icon={FiUser} placeholder="Nome" inputRef={register} error={errors.name} />`

Is there a default way to write something like

`<Input name="name" icon={FiUser} placeholder="Nome" aaa={bbb} />`

that would shorten it up? I will for now create a object that have both and only pass this object, but I want to know if there is a default way to do it, and if not, I strongly suggests that there should be! Something like regAndError on the useForm resulting object.

**Comment by bluebill1049** [comment, marked as answer]
`<Input name="name" icon={FiUser} placeholder="Nome" aaa={bbb} />`

you will have to `useFormContext` and register the input inside your component. 

https://react-hook-form.com/api#useFormContext

**Comment by ftzi** [comment]
Awesome! Just what I needed! Tried and works like a charm.

**Reply by ftzi** [reply]
Also, thanks for the quick answer!
